justfile

```diff
@@ -0,0 +1,284 @@
+        just pathver python .python-version
+    elif [[ -d conda ]]; then
+        source "$HOME/miniconda3/etc/profile.d/conda.sh"
+        conda activate "$(basename "$PWD")"
+        just pathver python .python-version
+    fi
+    if [[ -f .nvmrc ]]; then
+        just pathver node .nvmrc
+    fi
+
+# print PATH and VERsion; optionally assert version file matches
+pathver command version_file="":
+    #!/usr/bin/env bash
+    source=$(type -p "{{command}}")
+    if [[ -z $source ]]; then
+        source=$(type "{{command}}")
+    fi
+    actual_version=$("{{command}}" --version | sed -E "s/({{command}} )?//i")
+    echo "$source $actual_version"
+    if [[ -f "{{version_file}}" ]]; then
+        expected_version=$(cat "{{version_file}}")
+        if [[ $actual_version != "$expected_version" ]]; then
+            echo "ERROR: $source version $actual_version does not match {{version_file}} $expected_version"
+            exit 1
+        fi
+    fi
+
+# Docker Compose Build
+dcb *ARGS:
+    docker compose --progress=plain build {{ARGS}}
+
+# Docker Compose Push
+dcp *ARGS:
+    docker compose push --quiet {{ARGS}}
+
+# Docker Compose Run, respecting docker-compose.yaml port definitions
+dcr *ARGS:
+    docker compose run --quiet-pull --service-ports {{ARGS}}
+
+# Docker Compose Shell
+dcs *ARGS:
+    docker compose run "$(just cona)" bash {{ARGS}}
+
+# Docker Compose Up
+dcu *ARGS:
+    docker compose up {{ARGS}}
+
+# DEVelopment READYness check
+devready:
+    #!/usr/bin/env bash
+    set -euo pipefail
+    
+    grep -qE 'legacy_version_file.*=.*yes' ~/.asdfrc 2>/dev/null \
+        || echo 'WARNING: legacy_version_file != yes'
+    
+    installed=$(asdf plugin list 2>/dev/null)
+    [[ $installed == *nodejs* ]] || echo 'WARNING: nodejs plugin for asdf not added'
+    [[ $installed == *tenv* ]] || echo 'WARNING: tenv plugin for asdf not added'
+    [[ $installed == *uv* ]] || echo 'WARNING: uv plugin for asdf not added'
+    
+    # Git configs
+    [[ $(git config --global advice.skippedCherryPicks) == false ]] \
+        || echo 'WARNING: git advice.skippedCherryPicks != false'
+    [[ $(git config --global core.commentChar) == ';' ]] \
+        || echo 'WARNING: git core.commentChar != ;'
+    [[ $(git config --global diff.colormoved) == zebra ]] \
+        || echo 'WARNING: git diff.colormoved != zebra'
+    [[ $(git config --global user.name) ]] \
+        || echo 'ERROR: git user.name missing'
+    [[ $(git config --global user.email) ]] \
+        || echo 'ERROR: git user.email missing'
+    [[ $(git config --global pull.rebase) == true ]] \
+        || echo 'WARNING: git pull.rebase != true'
+    [[ $(git config --global push.default) == current ]] \
+        || echo 'WARNING: git push.default != current'
+    [[ $(git config --global rebase.autosquash) == true ]] \
+        || echo 'WARNING: git rebase.autosquash != true'
+
+# FORCE system to be READY for development, clobbering current settings
+forceready:
```

Comment:
I wonder about developing an bootstrap script that curls the latest version of this file, extracts this section, and runs it with bash.